indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/PartialHashSegmentGenerateTask.java

```diff
@@ -125,14 +127,18 @@ public boolean isReady(TaskActionClient taskActionClient) throws Exception
   }
 
   @Override
-  IndexTaskSegmentAllocator createSegmentAllocator(TaskToolbox toolbox) throws IOException
+  CachingSegmentAllocator createSegmentAllocator(TaskToolbox toolbox, ParallelIndexSupervisorTaskClient taskClient)
```

Comment:
Is it worth adding a unit test for this method now?

Reply:
Do you mean for `createSegmentAllocator()`? Is it covered by `HashPartitionMultiPhaseParallelIndexingTest`? BTW, the return type of this method will also be changed to `SegmentAllocator` in my follow-up PR.